commit 6149574dc0f9
Author: Gabriele Svelto <gsvelto@mozilla.com>
Date:   Tue Sep 26 09:35:03 2017 +0200

    Bug 1402966 - Search for libcurl in more paths to support various *BSDs. r=Dexter, a=sledru
    
    MozReview-Commit-ID: J4ykuSVEa0y
    
    --HG--
    extra : source : 4741b93cfdf29517ff8eae863825fdc2ab5bd7bd
---
 .../telemetry/pingsender/pingsender_unix_common.cpp      | 16 ++++++++++------
 1 file changed, 10 insertions(+), 6 deletions(-)

diff --git toolkit/components/telemetry/pingsender/pingsender_unix_common.cpp toolkit/components/telemetry/pingsender/pingsender_unix_common.cpp
index ae20f4114193..7817e93e3d1f 100644
--- toolkit/components/telemetry/pingsender/pingsender_unix_common.cpp
+++ toolkit/components/telemetry/pingsender/pingsender_unix_common.cpp
@@ -88,10 +88,19 @@ CurlWrapper::Init()
     "/usr/lib/i386-linux-gnu", // Debian 32-bit x86
     "/usr/lib/x86_64-linux-gnu", // Debian 64-bit x86
 #endif // XP_LINUX
+#if !defined(XP_MACOSX) && !defined(XP_LINUX) // Various BSDs
+    "/usr/local/lib", // FreeBSD, OpenBSD
+    "/usr/pkg/lib", // NetBSD
+#endif // !defined(XP_MACOSX) && !defined(XP_LINUX)
   };
 
   const char* libcurlNames[] = {
-#ifdef XP_LINUX
+#if defined(XP_MACOSX)
+    // macOS
+    "libcurl.dylib",
+    "libcurl.4.dylib",
+    "libcurl.3.dylib",
+#else // Linux, *BSD, ...
     "libcurl.so",
     "libcurl.so.4",
     // Debian gives libcurl a different name when it is built against GnuTLS
@@ -100,11 +109,6 @@ CurlWrapper::Init()
     // Older versions in case we find nothing better
     "libcurl.so.3",
     "libcurl-gnutls.so.3", // See above for Debian
-#elif defined(XP_MACOSX)
-    // macOS
-    "libcurl.dylib",
-    "libcurl.4.dylib",
-    "libcurl.3.dylib",
 #endif
   };
 
